test/realtime/resume.test.js

```diff
@@ -629,5 +622,121 @@ define(['shared_helper', 'async', 'chai'], function (helper, async, chai) {
+              function (cb) {
+                async.parallel(rxChannels.map((channel) => function (cb) {
+                  channel.attach(cb);
+                }), cb);
+              },
+              function (cb) {
+                function sendAndAwait(_, cb) {
+                  async.parallel(channelNames.map((name) => function (cb) {
+                    const tx = txRest.channels.get(name);
+                    const rx = rxRealtime.channels.get(name);
+                    const event = String(Math.random());
+                    async.parallel(
+                      [
+                        function (cb) {
+                          tx.publish(event, null, cb);
+                        },
+                        function (cb) {
+                          rx.subscribe(event, function (msg) {
+                            rx.unsubscribe(event);
+                            cb();
+                          });
+                        },
+                      ],
+                      cb
+                    )
+                  }), cb);
+                }
+                // Publish messages on each channel.
+                async.timesSeries(MSG_COUNT, sendAndAwait, cb);
+              },
+              function (cb) {
+                recoveryKey = rxRealtime.connection.getRecoveryKey();
+                helper.becomeSuspended(rxRealtime, cb);
+              },
+              function (cb) {
+                function publishAll(_, cb) {
+                  async.parallel(channelNames.map((name) => function (cb) {
+                    const tx = txRest.channels.get(name);
+                    tx.publish('sentWhileDisconnected', null, cb);
+                  }), cb);
+                }
+                // Publish another MSG_COUNT messages on each channel (which are no
+                // longer received by rxRealtime).
+                async.timesSeries(MSG_COUNT, publishAll, cb);
+              },
+              function (cb) {
+                rxRealtimeRecover = helper.AblyRealtime({ recover: recoveryKey });
+                rxRecoverChannels = channelNames.map((name) => rxRealtimeRecover.channels.get(name));
+                cb();
+              },
+              function (cb) {
+                // Subscribe to the first channels before becoming connected
+                // (so when the transport is activated the channels serials
+                // should be set).
+                async.parallel(rxRecoverChannels.slice(0, 5).map((channel) => function (cb) {
+                  let recoveredCount = 0;
+                  channel.subscribe('sentWhileDisconnected', function(msg) {
+                    recoveredCount++;
+                  });
+                  setTimeout(function () {
+                    try {
+                      expect(recoveredCount).to.equal(MSG_COUNT);
+                      cb();
+                    } catch (err) {
+                      cb(err);
+                    }
+                  }, 2000);
```

Comment:
Tests will become flaky because of the timeout, we can use [waitforexpect](https://www.npmjs.com/package/wait-for-expect) instead. It will move forward as soon as expectation is matched against given assertion.

Reply:
```
waitForExpect(() => {expect(numberToChange).toEqual(100)}).then(cb);
```

Reply:
We were using it in the `ably-echo` library -> https://github.com/ably-forks/echo/blob/afdf9c63ea81f2a8f54811d1a3e926577ea7dd2f/tests/ably/ably-channel.test.ts#L177